cli/main.rs

```diff
@@ -123,103 +126,212 @@ pub fn print_file_info(worker: &Worker, url: &str) {
+    // Setup runtime.
+    js_check(worker.execute("denoMain()"));
+    worker
+      .then(|result| {
         js_check(result);
         Ok(())
+      }).map_err(|(err, _worker): (RustOrJsError, Worker)| {
+        print_err_and_exit(err)
       })
-    });
-    tokio_util::run(main_future);
-  } else if let Some(main_module) = state.main_module() {
-    // Normal situation of executing a module.
-
-    let main_future = lazy(move || {
-      // Setup runtime.
-      js_check(worker.execute("denoMain()"));
-      debug!("main_module {}", main_module);
-
-      let main_url = root_specifier_to_url(&main_module).unwrap();
-
-      worker
-        .execute_mod_async(&main_url, should_prefetch)
-        .and_then(move |worker| {
-          if should_display_info {
-            // Display file info and exit. Do not run file
-            print_file_info(&worker, &main_module);
-            std::process::exit(0);
-          }
-          worker.then(|result| {
-            js_check(result);
-            Ok(())
-          })
-        }).map_err(|(err, _worker)| print_err_and_exit(err))
-    });
-    tokio_util::run(main_future);
-  } else {
-    // REPL situation.
-    let main_future = lazy(move || {
-      // Setup runtime.
-      js_check(worker.execute("denoMain()"));
-      worker
-        .then(|result| {
+  });
+  tokio_util::run(main_future);
+}
+
+fn run_script(flags: DenoFlags, argv: Vec<String>) {
+  let (mut worker, state) = get_worker_and_state(flags, argv);
+
+  let main_module = state.main_module().unwrap();
+  // Normal situation of executing a module.
+  let main_future = lazy(move || {
+    // Setup runtime.
+    js_check(worker.execute("denoMain()"));
+    debug!("main_module {}", main_module);
+
+    let main_url = root_specifier_to_url(&main_module).unwrap();
+
+    worker
+      .execute_mod_async(&main_url, false)
+      .and_then(move |worker| {
+        worker.then(|result| {
           js_check(result);
           Ok(())
-        }).map_err(|(err, _worker): (RustOrJsError, Worker)| {
-          print_err_and_exit(err)
         })
-    });
-    tokio_util::run(main_future);
+      }).map_err(|(err, _worker)| print_err_and_exit(err))
+  });
+  tokio_util::run(main_future);
+}
+
+fn fmt_command(mut flags: DenoFlags, mut argv: Vec<String>) {
+  argv.insert(1, "https://deno.land/std/prettier/main.ts".to_string());
+  flags.allow_read = true;
+  flags.allow_write = true;
+  run_script(flags, argv);
+}
+
+fn main() {
+  #[cfg(windows)]
+  ansi_term::enable_ansi_support().ok(); // For Windows 10
+
+  log::set_logger(&LOGGER).unwrap();
+  let args: Vec<String> = env::args().collect();
+  let cli_app = create_cli_app();
+  let matches = cli_app.get_matches_from(args);
+  let flags = flags::parse_flags(matches.clone());
+
+  if flags.v8_help {
+    // show v8 help and exit
+    // TODO(bartlomieju): this relies on `v8_set_flags` to swap `--v8-options` for "--help"
+    v8_set_flags(vec!["--v8-options".to_string()]);
```

Comment:
Other than this file only `core/examples/http_bench.rs` uses `v8_set_flags`. I guess we can now rid it of `v8_set_flags_preprocess` and remove this obscure swapping logic?

Reply:
@bartlomieju Nice clean ups. Getting rid `v8_set_flags_preprocess` seems good. Ping me when I should review again.

Reply:
Thanks @ry. This is now ready for review.